docs/merger_architecture.md

```diff
@@ -1,172 +0,0 @@
-# Merger architecture
-
-The merger is a new application that we are planning to write in Q3 2018.
-
-It receives source records/transformed Works, and it works out which source records correspond to the same Work in our pipeline.
-
-This document has some notes on the proposed architecture.
-
-## Model
-
-We model source records and their relationships as a graph.
-The vertices are the source records, and there is a directed edge from X to Y if information in the source record X tells us that it is related to Y.
-
-Each merged work is one of the connected components of this graph.
-
-![](relater_graph.png)
-
-In this example, there are three works:
-
--   A red work made of three source records
--   A green work made of two source records
--   A blue work that is a single source record
-
-The relater will receive updates to this graph, one vertex at a time.
-When it receives an update, it needs to tell us:
-
--   Any new works which have been created
--   Any old works which have been destroyed and need to be redirected (if two components emerge, or one component splits)
-
-## Storage
-
-We considered three storage options:
-
--   Neo4j (a filesystem-backed graph database)
--   Titan/JanusGraph (a DynamoDB-backed graph database)
--   Storing graph information in SQL
-
-We decided to use SQL, specifically RDS, as we already use it in the ID minter, and we know how it scales.
-
-The idea is to store the structure of the graph in SQL, but do all of the graph theory logic outside the database.
-Trying to do graph operations in SQL queries feels a step too far!
-
-## Database schema
-
-We can identify subgraphs/components by concatenating the sorted IDs of the vertices.
-For example, the graph below has three components: `A-B-C-D`, `E-F-G` and `H`.
-
-![](relater_component_ids.png)
-
-This is our current schema:
-
--   `source_component` -- contains the identifier of a component.
-    This may be a single vertex.
-
--   `tail_vertices` -- if the source component is a single vertex, a list of
-    vertices which are the "tail" of a directed edge from this vertex.
-
```

Comment:
Perhaps call these vertex head and tail, rather than source and tail.  Also, I'm not sure why this is a list of tail_vertices I thought this table was storing an edge per row?

Reply:
Uhm, yeah in hindsight that's not a correct description. It stores the single nodes and all the subgraphs that at some point have been advertised as a single work. I think it could work by storing an edge per row but it would just mean more operations when acquiring the locks, I'm not sure whether it would have any advantage